src/field_theory/complete_ordered.lean

```diff
@@ -0,0 +1,604 @@
+    { transitivity (0 : K),
+      push_neg at hq,
+      exact le_of_lt hq,
+      exact mul_self_nonneg (Sup (cut_image F K x)), }, },
+  { intros y hy,
+    by_cases hypos : 0 ≤ y,
+    { obtain ⟨q2, hqpos, hq21, hq22⟩ := exists_rat_sqr_btwn hy hypos,
+      rw pow_two at hq22,
+      have : (q2 : K) < _ := lt_of_mul_self_lt_mul_self _ _ hq22,
+      use (q2 : K)^2,
+      split,
+      norm_cast,
+      rw mem_cut_image_iff',
+      erw [induced_add_map, lt_induced_map_iff] at this,
+      obtain ⟨q3, hq23, hh⟩ := this,
+      rw pow_two,
+      push_cast,
+      have : (q2 : F) < x :=
+      begin
+        transitivity (q3 : F),
+        assumption_mod_cast,
+      end,
+      apply mul_lt_mul'' this this,
+      assumption_mod_cast,
+      assumption_mod_cast,
+      exact hq21,
+      exact_mod_cast hqpos,
+      simp only [induced_add_map, add_monoid_hom.coe_mk],
+      rw lt_induced_map_iff,
+      obtain ⟨q3, q30, q3x⟩ := exists_rat_btwn hpos,
+      use q3,
+      split,
+      assumption_mod_cast, },
+    { use ((0 : ℚ) : K),
+      split,
+      rw [mem_cut_image_iff', rat.cast_zero],
+      exact linear_ordered_field.mul_pos _ _ hpos hpos,
+      push_neg at hypos,
+      rw [rat.cast_zero],
+      exact hypos, }, }
+end
+
+@[simp]
+lemma induced_map_inv_self (F K : Type*) [conditionally_complete_linear_ordered_field F]
+  [conditionally_complete_linear_ordered_field K] (x : F) :
+  induced_map K F (induced_map F K x) = x :=
+begin
+  rw [induced_map],
+  apply cSup_intro (cut_image_nonempty K F (induced_map F K x)),
+  { intros y hy,
+    rw mem_cut_image_iff at hy,
+    rcases hy with ⟨q, rfl, h⟩,
+    rw induced_map at h,
+    obtain ⟨y, hym, hyq⟩ := exists_lt_of_lt_cSup (cut_image_nonempty F K x) h,
+    rw mem_cut_image_iff at hym,
+    obtain ⟨q2, rfl, h⟩ := hym,
+    apply le_of_lt,
+    transitivity (q2 : F),
+    assumption_mod_cast,
+    assumption, },
+  { intros w hw,
+    obtain ⟨q, hqw, hqx⟩ := exists_rat_btwn hw,
+    refine ⟨q, _, _⟩,
+    { rw [mem_cut_image_iff', lt_induced_map_iff],
+      obtain ⟨q2, hq2q, hq2x⟩ := exists_rat_btwn hqx,
+      refine ⟨q2, _, _⟩,
+      exact_mod_cast hq2q,
+      exact_mod_cast hq2x, },
+    { exact_mod_cast hqw, }, }
+end
+
+/-- The equivalence of ordered rings between two conditionally complete linearly ordered fields. -/
+def ordered_ring_equiv (F K : Type*)
+  [conditionally_complete_linear_ordered_field F] [conditionally_complete_linear_ordered_field K] :
+  ((≤) : F → F → Prop) ≃+*o ((≤) : K → K → Prop) :=
+{ to_fun := induced_map F K,
+  inv_fun := induced_map K F,
+  left_inv := induced_map_inv_self F K,
+  right_inv := induced_map_inv_self K F,
+  map_mul' := induced_map_mul F K,
+  map_add' := induced_map_add F K,
+  ord' := λ x y, begin
+    refine ⟨induced_map_le _ _, _⟩,
+    intro h,
+    replace h := induced_map_le K F h,
+    rw [equiv.coe_fn_mk, induced_map_inv_self, induced_map_inv_self] at h,
+    exact h,
+  end }
+
```

Comment:
Can you please add the fact that there is a *unique* such order preserving ring hom? In other words: the reals have no automorphisms...